Index: eglibc-2.17/debian/rules.d/build.mk
===================================================================
--- eglibc-2.17.orig/debian/rules.d/build.mk
+++ eglibc-2.17/debian/rules.d/build.mk
@@ -56,6 +56,8 @@
 	echo "libdir = $(call xx,libdir)"         >> $(DEB_BUILDDIR)/configparms
 	echo "slibdir = $(call xx,slibdir)"       >> $(DEB_BUILDDIR)/configparms
 	echo "rtlddir = $(call xx,rtlddir)"       >> $(DEB_BUILDDIR)/configparms
+	echo "have-ssp = no"                      >> $(DEB_BUILDDIR)/configparms
+	echo "config-extra-cflags = -fno-stack-protector"   >> $(DEB_BUILDDIR)/configparms
 
 	# Prevent autoconf from running unexpectedly by setting it to false.
 	# Also explicitly pass CC down - this is needed to get -m64 on
